src/EditorFeatures/TestUtilities/Classification/AbstractClassifierTests.cs

```diff
@@ -25,25 +24,42 @@ public abstract class AbstractClassifierTests
     {
         protected AbstractClassifierTests() { }
 
-        protected abstract Task<ImmutableArray<ClassifiedSpan>> GetClassificationSpansAsync(string text, TextSpan span, ParseOptions parseOptions);
+        protected abstract Task<ImmutableArray<ClassifiedSpan>> GetClassificationSpansAsync(string text, TextSpan span, ParseOptions parseOptions, bool outOfProcess);
 
         protected abstract string WrapInClass(string className, string code);
         protected abstract string WrapInExpression(string code);
         protected abstract string WrapInMethod(string className, string methodName, string code);
         protected abstract string WrapInNamespace(string code);
 
-        protected abstract Task DefaultTestAsync(string code, string allCode, FormattedClassification[] expected);
+        protected async Task DefaultTestAsync(string code, string allCode, FormattedClassification[] expected)
+        {
+            await DefaultTestAsync(code, allCode, outOfProcess: true, expected);
+            await DefaultTestAsync(code, allCode, outOfProcess: false, expected);
+        }
+
+        protected abstract Task DefaultTestAsync(string code, string allCode, bool outOfProcess, FormattedClassification[] expected);
+
+        protected async Task TestAsync(
+           string code,
+           string allCode,
+           ParseOptions parseOptions,
+           params FormattedClassification[] expected)
+        {
+            await TestAsync(code, allCode, parseOptions, outOfProcess: true, expected);
```

Comment:
> await TestAsync(code, allCode, parseOptions, outOfProcess: true, expected); [](start = 11, length = 76)

Is there much value in running *all* tests twice? Wouldn't it be sufficient to run all tests with oop=true and then just a single test that tests the fallback to local?

Reply:
When we did OOP for FAR/NavTo we found many bugs in both directions as we had to change things.  so i'm a fan of actually running *everything* through both paths.

Reply:
Note: for classification it may be less valuable since the effort is minimal.  but this also only adds a couple of seconds of testing, so i don't mind having this.

Reply:
I don't understand how that could be. The remote case = local case + serialization. 

---
In reply to: [385928992](https://github.com/dotnet/roslyn/pull/41998#discussion_r385928992) [](ancestors = 385928992)

Reply:
in the case of classification, that's true.  in the case of navto/far it was not.  for example, we had literal different implementations of certain types (i.e. ones backed by symbols, versus ones backed by raw data).  Issues can and did arise.  At the time we didn't want to take the code of having the inproc system use the same *data types* as the out of proc system a that could generate a lot of in-proc CPU and memory to convert over.  This impacted user experience in a way that OOP did not (since it was literally not fighting for the same memory resources as VS).